src/assets/CollectiblesController.ts

```diff
@@ -1045,14 +1044,25 @@ export class CollectiblesController extends BaseController<
    *
    * @param collectible - The collectible object to check and update.
    * @param batch - A boolean indicating whether this method is being called as part of a batch or single update.
+   * @param accountParams - (optional) - The userAddress and chainId to check ownership against
    * @returns the collectible with the updated isCurrentlyOwned value
    */
   async checkAndUpdateSingleCollectibleOwnershipStatus(
     collectible: Collectible,
     batch: boolean,
+    accountParams?: AccountParams,
   ) {
     const { allCollectibles } = this.state;
-    const { selectedAddress, chainId } = this.config;
+
+    let selectedAddress, chainId;
+    // if accountParams is defined we use those values for address and chainId
+    // if not we use the currently configured values for address and chainId
+    if (accountParams) {
+      selectedAddress = accountParams.userAddress;
+      chainId = accountParams.chainId;
+    } else {
+      ({ selectedAddress, chainId } = this.config);
+    }
```

Comment:
[nit] I think you can simplify this as follows:

``` suggestion
    { userAddress: selectedAddress, chainId }: AccountParams | undefined = {
      userAddress: this.config.selectedAddress,
      chainId: this.config.chainId
    }
  ) {
    const { allCollectibles } = this.state;
```

Reply:
updated here: https://github.com/MetaMask/controllers/pull/672/commits/0dcc89542336640a1f8dccb2cd9bca82b4c1dd03